### PR TITLE
ci: Update MacOS runner to macos13

### PR DESCRIPTION
**Pull request by aleksraiden**
Due to https://github.blog/changelog/2024-05-20-actions-upcoming-changes-to-github-hosted-macos-runners/   we are change macos11 up to next, macos12

### CODE REVIEW

**Comment by PragmaTwice** [comment]
Need to confirm why there are go test failures on macos 12.

**Comment by aleksraiden** [comment]
realpath util not included at macos-12 (see: https://apple.stackexchange.com/questions/450035/is-the-unix-realpath-command-distributed-with-macos-ventura)

**Comment by aleksraiden** [comment]
realpath are fixed, some go integration test broken by i/o timeout

File: .github/workflows/kvrocks.yaml [hunk 1]
```diff
@@ -111,21 +111,21 @@ jobs:
       matrix:
         include:
           - name: Darwin Clang
-            os: macos-11
+            os: macos-12
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
            os: macos-13
```

File: .github/workflows/kvrocks.yaml [hunk 2]
```diff
@@ -111,21 +111,21 @@ jobs:
...
             compiler: auto
           - name: Darwin Clang arm64
             os: macos-14
             compiler: auto
           - name: Darwin Clang without Jemalloc
-            os: macos-11
+            os: macos-12
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
            os: macos-13
```

File: .github/workflows/kvrocks.yaml [hunk 3]
```diff
@@ -111,21 +111,21 @@ jobs:
...
             compiler: auto
             without_jemalloc: -DDISABLE_JEMALLOC=ON
           - name: Darwin Clang with OpenSSL
-            os: macos-11
+            os: macos-12
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
            os: macos-13
```

File: .github/workflows/kvrocks.yaml [hunk 4]
```diff
@@ -111,21 +111,21 @@ jobs:
...
             compiler: auto
             with_openssl: -DENABLE_OPENSSL=ON
           - name: Darwin Clang without luaJIT
-            os: macos-11
+            os: macos-12
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
            os: macos-13
```

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**